src/test/java/teammates/test/pageobjects/AppPage.java

```diff
@@ -901,12 +897,8 @@ private static String replaceUnpredictableValuesWithPlaceholders(String content)
                 // today's date
                 .replace(TimeHelper.formatDate(now).replace("/", "&#x2f;"), "${today}")
                 .replace(TimeHelper.formatDate(now), "${today}")
-                // now (used in comments last edited date) e.g. [Thu, 07 May 2015, 07:52:13 UTC]
-                .replaceAll(new SimpleDateFormat("EEE, dd MMM yyyy, ").format(now) + "[0-9]{2}:[0-9]{2}:[0-9]{2} UTC", "\\${comment\\.date}")
-                // now (date, time)
-                .replaceAll(new SimpleDateFormat("EEE, dd MMM yyyy, ").format(now) + "[0-9]{2}:[0-9]{2} [AP]M", "\\${datetime\\.now}")
-                // dynamic feedback submission numbers
-                .replaceAll("(?s)<span class=\"submissionsNumber\".*?</span>", "<span class=\"submissionsNumber\" id=\"submissionsNumber\">\\${submissions\\.number}</span>")
+                // now (date, time) e.g. [Thu, 07 May 2015, 07:52 PM] or [Thu, 07 May 2015, 07:52:13 UTC]
+                .replaceAll(new SimpleDateFormat("EEE, dd MMM yyyy, ").format(now) + "[0-9]{2}:[0-9]{2}( [AP]M|:[0-9]{2} UTC)", "\\${datetime\\.now}")
```

Comment:
May be my understanding of this situation is incomplete from the beginning. Let's take the truncation example. Let's say there are 8 places in the UI where we use truncated Google ID and 10 places where we use the full ID. Now let's say a certain PR incorrectly changed a jsp page such that one of those 8 places is now using full ID but no html files used for testing was changed by the PR. Will that result in a test failure because the test expects the truncated ID but the actual html has the full ID?
If the answer is 'no', that is, a dev can change the jsp page without changing the html page and still pass the test cases, then our tests are not as strong as they can be. That is, the tests accepts either full ID or truncated ID and doesn't care which one is given.
My understanding is, in our current code, a dev can change a jsp page from 'truncated' to 'full' (or vice versa) without changing html pages, and still pass the tests. Am I right?

Similarly, let's assume we have two date formats. (Although you have now standardized it to one format, it is quite likely we will have multiple formats in future. e.g. a short format and a long format.)
If a dev can change a jsp from one format to the other, and still pass the tests (without changing any html files), then the tests are weak. If the tests are strong, any change to UI should be accompanied by a corresponding changes to test html files.

Now, if we are weakening our tests in the above manner, it should be weighed against the performance (or any other) gains we are getting in return.